advanced/index.php

```diff
@@ -1,224 +1,335 @@
-// If the server name is 'pi.hole', it's likely a user trying to get to the admin panel.
-// Let's be nice and redirect them.
-if ($serverName === 'pi.hole')
-{
-    header('HTTP/1.1 301 Moved Permanently');
-    header("Location: /admin/");
+// Set landing page name, found within /var/www/html/pihole/
+$landPage = "../landing.php";
+
+// Set empty array for hostnames to be accepted as self address for splash page
+$authorizedHosts = [];
+
+// Append FQDN to $authorizedHosts
+if (!empty($svFQDN)) array_push($authorizedHosts, $svFQDN);
+  
+// Append virtual hostname to $authorizedHosts
+if (!empty($_SERVER["VIRTUAL_HOST"])) {
+    if (validate_server_name($_SERVER["VIRTUAL_HOST"]) === TRUE) {
+        array_push($authorizedHosts, $_SERVER["VIRTUAL_HOST"]);
+    } else {
+        die("[ERROR]: <code>VIRTUAL_HOST</code> header output does not appear to be valid: <code>".$_SERVER["VIRTUAL_HOST"]."</code>");
+    }
 }
 
-// Retrieve server URI extension (EG: jpg, exe, php)
-ini_set('pcre.recursion_limit',100);
-$uriExt = pathinfo($uri, PATHINFO_EXTENSION);
+// Set which extension types get rendered as "Website Blocked" (Including "" for index file extensions)
+$validExtTypes = array("asp", "htm", "html", "php", "rss", "xml", "");
 
-// Define which URL extensions get rendered as "Website Blocked"
-$webExt = array('asp', 'htm', 'html', 'php', 'rss', 'xml');
+// Get extension of current URL
+$currentUrlExt = pathinfo($_SERVER["REQUEST_URI"], PATHINFO_EXTENSION);
 
-// Get IPv4 and IPv6 addresses from setupVars.conf (if available)
-$setupVars = parse_ini_file("/etc/pihole/setupVars.conf");
-$ipv4 = isset($setupVars["IPV4_ADDRESS"]) ? explode("/", $setupVars["IPV4_ADDRESS"])[0] : $_SERVER['SERVER_ADDR'];
-$ipv6 = isset($setupVars["IPV6_ADDRESS"]) ? explode("/", $setupVars["IPV6_ADDRESS"])[0] : $_SERVER['SERVER_ADDR'];
-
-$AUTHORIZED_HOSTNAMES = array(
-	$ipv4,
-	$ipv6,
-	str_replace(array("[","]"), array("",""), $_SERVER["SERVER_ADDR"]),
-	"pi.hole",
-	"localhost");
-// Allow user set virtual hostnames
-$virtual_host = getenv('VIRTUAL_HOST');
-if (!empty($virtual_host))
-	array_push($AUTHORIZED_HOSTNAMES, $virtual_host);
-
-// Immediately quit since we didn't block this page (the IP address or pi.hole is explicitly requested)
-if(validIP($serverName) || in_array($serverName,$AUTHORIZED_HOSTNAMES))
-{
-	http_response_code(404);
-	die();
+// Set mobile friendly viewport
+$viewPort = '<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>';
+
+// Set response header
+function setHeader($type = "x") {
+    header("X-Pi-hole: A black hole for Internet advertisements.");
+    if (isset($type) && $type === "js") header("Content-Type: application/javascript");
 }
 
-if(in_array($uriExt, $webExt) || empty($uriExt))
-{
-	// Requested resource has an extension listed in $webExt
-	// or no extension (index access to some folder incl. the root dir)
-	$showPage = true;
+// Determine block page redirect
+if ($serverName === "pi.hole") {
```

Comment:
So the logic is that we need to determine if the website that's being visited is one for the Pi, or one on a block list.

Virtual Host will redirect to the splash page (Local Pi landing page) by [L42](https://github.com/WaLLy3K/pi-hole/blob/69fa9874dd2a05e83225220fd46aa0a425f32aa3/advanced/index.php#L42), while everything else is covered by checking if it's a valid IP in [L68](https://github.com/WaLLy3K/pi-hole/blob/69fa9874dd2a05e83225220fd46aa0a425f32aa3/advanced/index.php#L68).